utils/symbolicate-linux-fatal.py

```diff
@@ -0,0 +1,119 @@
+#!/usr/bin/env python
+
+# TODOs:
+# * verbose output
+# * search symbols by name for the not <unavailable> ones
+
+import sys
+import lldb
+from subprocess import Popen, PIPE
+
+dynlibs = {}
+def process_ldd(lddoutput):
+    global dynlibs
+    for line in lddoutput.splitlines():
+        ldd_tokens = line.split()
+        if len(ldd_tokens) >= 3:
+            dynlibs[ldd_tokens[0]] = ldd_tokens[2]
+
+def create_lldb_target(memmap):
+    lldb_target = lldb_debugger.CreateTargetWithFileAndArch(binary, lldb.LLDB_ARCH_DEFAULT)
+    module = lldb_target.GetModuleAtIndex(0)
+    # lldb seems to treat main binary differently, slide offset must be zero
+    lldb_target.SetModuleLoadAddress(module, 0)
+    for dynlib_path in memmap:
+        if not binary in dynlib_path:
+            module = lldb_target.AddModule (dynlib_path, lldb.LLDB_ARCH_DEFAULT, None, None)
+            lldb_target.SetModuleLoadAddress(module, memmap[dynlib_path])
+    return lldb_target
+
+def process_stack(stack):
+    global binary
+    if len(stack) == 0:
+        return
+    memmap = {}
+    full_stack = []
+    for line in stack:
+        stack_tokens = line.split()
+        dynlib_fname = stack_tokens[1]
+        if dynlib_fname in dynlibs:
+            dynlib_path = dynlibs[dynlib_fname]
+        else:
+            if dynlib_fname in binary:
+                dynlib_path = binary
+            else:
+                dynlib_path = None
+
+        if "<unavailable>" in stack_tokens[3]:
+            framePC = int(stack_tokens[2], 16)
+            symbol_offset = int(stack_tokens[-1], 10)
+            dynlib_baseaddr = framePC - symbol_offset
+            if dynlib_path in memmap:
+                if memmap[dynlib_path] != dynlib_baseaddr:
+                    print "Mismatched base address for: {0:s}, had: {1:x}, now got {2:x}".format(dynlib_path, memmap[dynlib_path], dynlib_baseaddr)
```

Comment:
To make this script compatible with Python 3, you should `from __future__ import print_function` above, then use `print()` here.